### ISSUE TITLE
test and finalize TaskPublish in non DryMode

### ISSUE THREAD

**Issue by belforte** [dmwm/CRABServer#6100]
capitalizing on closure of  #5832 which was getting too long to follow. 
so will start from master branch. in a cleaner situation.

**Comment by belforte** [comment]
when running ijnteractively with pdb, redirect logger to console.

**Comment by belforte** [comment]
- [ ] need a way to cleanup files in /data/srv/Publisher_files

**Comment by belforte** [comment]
- [ ] since now it waits for Nslaves to be complted before submitting more, one task taking long (.eg. large migration) will block everything, need to somehow always keep Nslaves active. W/o the current `process.join()` which blocks until all subprocesses have ended.

**Comment by belforte** [comment]
- [ ]  some serious problem. master log is full of
```
2020-06-30 04:33:25,899:ERROR:master:Error during metadata retrieving from /crabserver/preprod/filemetadata: url=https://cmsweb-testbed.cern.ch/crabserver/preprod/filemetadata, code=414, reason=Request-URI Too Long, headers={'Date': 'Tue, 30 Jun 2020 02:33:25 GMT', 'Content-Length': '248', 'Content-Type': 'text/html; charset=iso-8859-1', 'Connection': 'close', 'Server': 'Apache'}
```
which immediately follow
```
2020-06-30 04:33:20,907:INFO:master:Skipped 181210_163217:belforte_crab_20181210_173203. Ignore tasks created before 2020.
```

**Comment by belforte** [comment]
OK. in the comment above, about who logs, the thing is that the `startSlave` function does not define `self.logger` but only `logger`, so when it calls `self.getPublDescFiles` the logger used in there is `self.logger` i.e. the master one.
-  [ ] Basically the use of logger and self requires a good cleanup.

**Comment by belforte** [comment]
- [ ] should not update publication time if in dry mode nor if all publication attempt failed

**Comment by belforte** [comment]
- [ ] log some progress report during mark_good and mark_failed becasue they are AWFULLY slow

**Comment by belforte** [comment]
This has to too long and confused.
I will push current status to master and open new issues for each of the still open points in comments above and possible new things. All in all this "stinks", but works.

**Comment by belforte** [comment]
closed via #6125
